base/type_intersect_exact.jl

```diff
@@ -0,0 +1,98 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module TypeIntersectExact
+
+struct OK end
+
+"""
+When `OK <: ok`, `R` is the exact type intersection. When `ok <: Union{}`, the
+exact type intersection wasn't found and `R` has no significance.
+"""
+struct Result{R, ok<:OK} end
```

Comment:
The fact that this is implementing an Option type seems pretty sketchy to me.

Reply:
How is it sketchy? It just seems like an elegant encoding to me?

Reply:
The idea is to let the caller know if the exact intersection couldn't be found.

Reply:
that's relatively sensible, the weird parts are that
1. the OK value is a type? Why not a `bool`?
2. Implimenting a custom option type specifically for this functionality seems somewhat weird. Why is an option type inherently embedded in the implementation of a `typeintersect` wrapper?

Reply:
it may be elegant, but something like that should be a separate PR, and not live under `TypeIntersectExact` namespace.

Reply:
Now that I think about this again, `OK` isn't necessary at all, and thus `Result` is also unnecessary: I think it would be better to just return a `Type` when the search was successful and `nothing` when the search failed.

Reply:
@adienes what do you mean by "not live under `TypeIntersectExact` namespace"? Do you object to the name, or to the addition of a namespace?

Reply:
Do you have a suggestion for a better name?

Reply:
I mean an `Optional` type has broader use than for this specific utility